integration-tests/suites/base.go

```diff
@@ -128,6 +128,9 @@ func (s *IntegrationTestSuiteBase) StopCollector() {
 func (s *IntegrationTestSuiteBase) Collector() collector.Manager {
 	if s.collector == nil {
 		s.collector = collector.New(s.Executor(), s.T().Name())
+	} else {
```

Comment:
[nitpick] Just to prevent constantly copying the same name into the testName when we are trying to access the collector object. Alternatively, this could be checked in the setter itself.
```suggestion
	} else if collector.testName != s.T().Name() {
```